en/ros2/px4_sdk.md

```diff
@@ -0,0 +1,336 @@
+Motors and servos can be set independently. Be aware that the assignment is vehicle and setup-specific.
+For example to control a quadrotor, you need to set the first 4 motors according to its [output configuration](../concept/control_allocation.md).
+
+:::note
+If you want to control an actuator that does not control the vehicle's motion, but for example a payload servo, see [below](#controlling-an-independent-actuator-servo).
+:::
+
+### Controlling an Independent Actuator/Servo
+
+If you want to control an independent actuator (a servo), follow these steps:
+
+1. [Configure the output](../payloads/#generic-actuator-control-with-mavlink)
+2. Create an instance of [px4_sdk::OffboardActuatorControls](https://github.com/PX4/px4_sdk/blob/main/px4_sdk_cpp/include/px4_sdk/control/offboard_actuators.hpp) in the constructor of your mode
+3. Call the `set` method to control the actuator(s). This can be done independently of any active setpoints.
+
+### Telemetry
+
+Telemetry, such as local or global position estimates can be found under [px4_sdk/odometry](https://github.com/PX4/px4_sdk/tree/main/px4_sdk_cpp/include/px4_sdk/odometry).
+
+### Failsafes and Mode Requirements
+
+Each mode has a set of requirement flags. These are generally automatically set, depending on which objects are used within the context of a mode. For example when adding manual control input with
+
+```cpp
+_manual_control_input = std::make_shared<px4_sdk::ManualControlInput>(*this);
+```
+
+the requirement flag for manual control gets set.
+Specifically, setting a flag has the following consequences in PX4, if the condition is not met:
+
+- arming is not allowed, while the mode is selected
+- when already armed, the mode cannot be selected
+- when armed and the mode is selected, the relevant failsafe is triggered (e.g. RC loss for the manual control requirement).
+  Check the [safety page](../config/safety.md) for how to configure failsafe behavior.
+  A failsafe is also triggered when the mode crashes or becomes unresponsive while it is selected.
+
+It is possible to manually update any mode requirement after the mode is registered. For example to add home position as requirement:
+
+```cpp
+modeRequirements().home_position = true;
+```
+
+#### Deferring Failsafes
+
+A mode or mode executor can temporarily defer non-essential failsafes.
+To do so, use the method `deferFailsafesSync`. And to get notified when a failsafe would be triggered, override the method `void onFailsafeDeferred()`.
+Check the [integration test](https://github.com/PX4/px4_sdk/blob/main/px4_sdk_cpp/test/integration/overrides.cpp#L167) for an example.
+
+### Assigning a Mode to an RC Switch or Joystick Action
+
+External modes can be assigned to RC switches or joystick actions.
+When assigning a mode to an RC switch, you need to know the index.
```

Comment:
Why? Don't you just need the name?
To put it another way, the mode assignment shows a distinct name, so when/why do you need the index?

Reply:
The parameter metadata does not contain the dynamic mode name

Reply:
It might be worth saying "why" in the docs, unless you already did so.